ci/migrate.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"context"
+
+	"github.com/openmeterio/openmeter/ci/internal/dagger"
+	"github.com/sourcegraph/conc/pool"
+)
+
+func (m *Ci) Migrate() *Migrate {
+	return &Migrate{
+		Source: m.Source,
+	}
+}
+
+type Migrate struct {
+	Source *dagger.Directory
+}
+
+func (m *Migrate) Check(
+	ctx context.Context,
+) error {
+	nix := nix(m.Source)
+	p := pool.New().WithErrors().WithContext(ctx)
+
+	// Always validate schema is generated
+	p.Go(syncFunc(nix.
+		WithExec([]string{"sh", "-c", "nix develop --impure .#dagger -c go generate ./internal/ent/..."}).
+		WithExec([]string{"sh", "-c", "nix develop --impure .#dagger -c sh -c 'if [[ -n $(git status --porcelain ./internal/ent) ]]; then echo \"Ent schema wasnt generated\"; exit 1; fi' "}),
```

Comment:
You can either use Git here, but you could also try using `dagger.Directory.Diff` to see if there is a diff between the first and the last version. That would utilize caching more.